tests/integration/abi_test.py

```diff
@@ -90,3 +97,112 @@ def test_dynamic_array_sum():
         args, "last log messed up"
     )
     assert inspector.stack_top() == 1, inspector.report(args, "stack top messed up")
+
+
+def process_filename(filename):
+    abi_info = filename.split("/")[-1].split(".")[0].split("_")[2:]
+
+    length = None
+    if len(abi_info) > 1:
+        abi_str, length = abi_info
+        length = int(length[1:-1])
+    else:
+        abi_str = abi_info[0]
+
+    abi_instance = abi.ABIType.from_string(abi_str)
+    abi_strat = ABIStrategy(abi_instance, length)
+    return abi_str, length, abi_instance, abi_strat
+
+
+def get_roundtrip_teals():
+    return list(ROUNDTRIP.glob("*.teal"))
+
+
+@pytest.mark.parametrize("roundtrip_app", get_roundtrip_teals())
```

Comment:
@tzaffi I'm probably missing some intuition.  Can you help me understand what's generating the _rountrip_app_ tests?

I'm trying to understand the workflow a maintainer will go through when the tests requiring updates.

Reply:
Admittedly this is murky and violates "explicit is preferred over implicit". `get_roundtrip_teals()` scans `tests/teal/roundtrip/` for all `*.teal` files and then parse the filename to see what the corresponding ABI-type should be. 

Should I follow a more explicit approach that starts with the filenames, rather than reading the directory?

Reply:
@tzaffi Sorry, I'm asking:  How are the .teal sources generated?

I don't see an issue with listing the dir for tests because the dir is specialized for _roundtrip_app_ tests.

Reply:
These tests were[ generated by PyTeal here](https://github.com/algorand/pyteal/blob/1ee615117da7617078fa2d68ad8546966a734eb7/tests/integration/abi_roundtrip_test.py#L100), and then I copied them manually into this repo.